catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/web/metrics/IcebergMetricsManager.java

```diff
@@ -0,0 +1,186 @@
+      } catch (InterruptedException e) {
+        LOG.warn("Iceberg Metrics writer thread is interrupted.");
+        break;
+      }
+      if (metricsReport != null) {
+        doSave(metricsReport);
+      }
+    }
+
+    MetricsReport metricsReport = queue.poll();
+    while (metricsReport != null) {
+      logMetrics("Drop Iceberg metrics because it's time to close metrics store.", metricsReport);
+      metricsReport = queue.poll();
+    }
+  }
+
+  private IcebergMetricsStore loadIcebergMetricsStore(String metricsStoreName) {
+    String metricsStoreClass =
+        ICEBERG_METRICS_STORE_NAMES.getOrDefault(metricsStoreName, metricsStoreName);
+    LOG.info("Load Iceberg metrics store: {}.", metricsStoreClass);
+    try {
+      return (IcebergMetricsStore)
+          Class.forName(metricsStoreClass).getDeclaredConstructor().newInstance();
+    } catch (Exception e) {
+      LOG.error(
+          "Failed to create and initialize Iceberg metrics store by name {}.", metricsStoreName, e);
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void logMetrics(String message, MetricsReport metricsReport) {
+    LOG.info("{} {}.", message, icebergMetricsFormatter.toPrintableString(metricsReport));
+  }
+
+  public void start() {
+    metricsWriterThread.start();
+    metricsCleanerExecutor.ifPresent(
+        executorService ->
+            executorService.scheduleAtFixedRate(
+                () -> {
+                  Instant now = Instant.now();
+                  Instant expireTime =
+                      IcebergRestUtils.calculateNewTimestamp(now, -24 * retainDays);
+                  LOG.info("Try clean Iceberg expired metrics, {}.", expireTime);
+                  try {
+                    icebergMetricsStore.clean(expireTime);
+                  } catch (Exception e) {
+                    LOG.warn("Clean Iceberg metrics failed.", e);
+                  }
+                },
+                0,
+                1,
+                TimeUnit.HOURS));
+  }
+
+  private void doSave(MetricsReport metricsReport) {
+    try {
+      icebergMetricsStore.save(metricsReport);
+    } catch (Exception e) {
+      LOG.warn("Write Iceberg metrics failed.", e);
+    }
+  }
+
+  public void save(MetricsReport metricsReport) {
+    if (isClosed) {
+      logMetrics("Drop Iceberg metrics because Iceberg Metrics Manager is closed.", metricsReport);
+      return;
+    }
+    if (queue.offer(metricsReport) == false) {
+      logMetrics("Drop Iceberg metrics because metrics queue is full.", metricsReport);
+    }
+  }
+
+  public void close() {
+    isClosed = true;
+    metricsCleanerExecutor.ifPresent(executorService -> executorService.shutdownNow());
+
+    if (metricsWriterThread != null) {
+      metricsWriterThread.interrupt();
+      try {
+        metricsWriterThread.join();
+      } catch (InterruptedException e) {
+        LOG.warn("Iceberg metrics manager is interrupted while join metrics writer thread.");
+        return;
```

Comment:
You should not return directly here, you still need to execute the follow-up logic.

Reply:
If IcebergMetricsManager is interrupted, I think it's better to quit as quickly as possible in case the other cleanup actions block the close.

Reply:
At least you should close the iceberg metrics store, right?

Reply:
done